java/arcs/sdk/android/storage/ServiceStore.kt

```diff
@@ -188,9 +189,13 @@ class ServiceStore<Data : CrdtData, Op : CrdtOperation, ConsumerData>(
     override suspend fun onProxyMessage(message: ProxyMessage<Data, Op, ConsumerData>): Boolean {
         val service = checkNotNull(storageService)
         val result = DeferredResult(coroutineContext)
+        // Trick: make an indirect access to the message to keep kotlin flow
+        // from holding the entire message that might encapsulate a large size data.
+        val messageRef: AtomicRef<ProxyMessage<Data, Op, ConsumerData>?> = atomic(message)
```

Comment:
is an atomic ref even needed? If you used a mutable local like

var messageCopy = message

and then messageCopy = null in the lambda block, wouldn't it achieve the same effect?

Reply:
Good point, never tried, lemme give it a shot.

Reply:
Using `var` works! Thanks.